### PR TITLE
[10.x.x] Fixed GLES2 compilation errors

### PR DESCRIPTION
**Pull request by phi-lira**
### Purpose of this PR
 - Removed `EncodeDepthNormal` function as that was not being used and it was causing compilation issues in GLES2 due to FloatPacking functions not being supported in that platform.
 - Removed gles2, gles3 and glcore shader target from GBuffer passes. These were also causing shader compilation erros in GLES2 and deferred renderer is already disabled for those platforms.
  - Post-processing was already disabled for GLES2, however several of PP shaders were still compiling for it and some had errors related to SV_VertexID instrisic.

---
### Testing status

**Manual Tests**: Tested building to Android phone using GLES2. There are still some shader compilation errors in GLES2 that seem related to XR. They are aware of that now and this should be fixed in a follow up PR.

- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: There's no GLES2 automation yet.

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs [hunk 1]
```diff
@@ -148,12 +150,20 @@ private enum ShaderPasses
             internal ScreenSpaceAmbientOcclusionPass()
             {
                 m_CurrentSettings = new ScreenSpaceAmbientOcclusionSettings();
+                if (SystemInfo.graphicsDeviceType == GraphicsDeviceType.OpenGLES2)
+                    m_IsDeviceGLES2 = true;
```
Comment (Chman) [inline com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs:154]: Given that GLES2 hardware will have a hard time running SSAO (same as PostFX), I wonder if it's useful to even go through the trouble of supporting it there...

File: com.unity.render-pipelines.universal/Shaders/PostProcessing/GaussianDepthOfField.shader [hunk 2]
```diff
@@ -3,6 +3,7 @@ Shader "Hidden/Universal Render Pipeline/GaussianDepthOfField"
     HLSLINCLUDE
 
         #pragma target 3.5
+        #pragma exclude_renderers gles
```
Comment (Chman) [inline com.unity.render-pipelines.universal/Shaders/PostProcessing/GaussianDepthOfField.shader:6]: Redundant with `#pragma target 3.5` but it doesn't hurt anyway.

File: com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs (hunk 1 above)
Comment (phi-lira) [reply com.unity.render-pipelines.universal/Runtime/RendererFeatures/ScreenSpaceAmbientOcclusion.cs:154]: Will let @ellioman comment on that when he's back. I agree, but then we should warn nicely in the UI.

**Comment by phi-lira** [comment]
Failures related to HDRP

**Pull request merged**